### ISSUE TITLE
Exclude folders in gtd 

### ISSUE THREAD

**Issue by jakobkhansen** [nvim-neorg/neorg#269]
### Issues

- [X] I have checked [existing issues](https://github.com/vhyrro/neorg/issues?q=is%3Aissue) and there are no existing ones with the same request.

### Feature description

Hi,

it would be nice if one could exclude entire folders (recursively) in `core.gtd.base`. I for example want to exclude everything in my `journals/` folder.

### Help

No

**Comment by danymat** [comment]
Hello, can you try out the implementation ?

**Comment by jakobkhansen** [comment]
Yup looks like it's working with files directly inside the directory. However, would it be possible to recursively exclude all nested files in the folder? For example the journal folder has many subdirectories and files, and these are not excluded if I exclude the journal folder.

**Comment by danymat** [comment]
Hmm are you sure? When adding `journal` in `exclude`, I get all files excluded from it

**Comment by jakobkhansen** [comment]
Very weird, I created a new workspace and there it works fine, but in my usual workspace with many other projects, it still shows up. Seems like a problem on my end, I'll try to figure it out :smile:

**Comment by danymat** [comment]
Hello, can you try enabling logging in your Neorg setup:

```
setup({
load=...,
	logger = {
		level = "info", -- Show trace, info, warning, error and fatal messages
	},
})

And try again? it'll show every file that will be parsed for GTD.

**Comment by jakobkhansen** [comment]
The log gave this output:

```
[neorg] [INFO  16:38:13] .../neorg/lua/neorg/modules/core/gtd/queries/retrievers.lua:93: files being parse
d for GTD:  { "index.norg", "projects-personal.norg", "projects-work.norg", "journal/2021/12/21.norg", "jo
urnal/2021/12/23.norg", "journal/2021/12/28.norg" }                                                       
[neorg] [INFO  16:38:13] .../neorg/lua/neorg/modules/core/gtd/queries/retrievers.lua:93: files being parse
d for GTD:  { "index.norg", "projects-personal.norg", "projects-work.norg", "journal/2021/12/21.norg", "jo
urnal/2021/12/23.norg", "journal/2021/12/28.norg" }
```

My gtd config looks like this:

```lua
["core.gtd.ui"] = {},
["core.gtd.queries"] = {},
["core.gtd.base"] = {
	config = {
		workspace = "gtd",
		default_lists = {
			inbox = "inbox.norg",
		},
		exclude = {
			"journal",
		},
	},
},
["core.norg.journal"] = {
	config = {
		workspace = "gtd",
	},
},
```

It seems like it is still parsing some files in the journal folder, but it is correctly ignoring some.

**Comment by danymat** [comment]
Hello, can you try again ?

**Comment by jakobkhansen** [comment]
Great, seems like it works now :smile:

**Comment by jakobkhansen** [comment]
Hello, I just noticed that projects in excluded folders still show up when _capturing_ a task using `:Neorg gtd capture` and pressing `p -> Add to project`. Maybe this listing of projects does not exclude here like you fixed for the project view? I would expect the exclude option to also exclude when capturing a task, right?

**Comment by danymat** [comment]
Right, it seems that I overlooked this.
I reopen this PR and provide a fix as soon as I can.

**Comment by danymat** [comment]
Hello, I fixed it in 8ac21bdd. Can you try out ?

**Comment by jakobkhansen** [comment]
Yes, seems like it works for me, thanks!
